activemodel/lib/active_model/type/float.rb

```diff
@@ -7,6 +7,15 @@ def type
         :float
       end
 
+      def type_cast_for_schema(value)
+        return "::Float::NAN" if value.try(:nan?)
```

Comment:
This line makes me angry that the decision on how to interpret "NaN does not equal to NaN" should also apply to case equality